pkg/storage/unified/resource/server.go

```diff
@@ -285,7 +286,7 @@ func (s *server) newEvent(ctx context.Context, user claims.AuthInfo, key *Resour
 	}
 
 	if obj.GetUID() == "" {
-		s.log.Error("object is missing UID", "key", key)
+		return nil, NewBadRequestError("object is missing UID")
```

Comment:
@chaudyg if my assumption about deployment is wrong or dangerous, we can should comment this out and say "TODO, uncomment soon"